### PR TITLE
[gardening] Remove unused code, fix typos and improve file header formatting

### PR DESCRIPTION
**Pull request by practicalswift**
#### What's in this pull request?

Remove unused code, fix typos and improve file header formatting.
#### Resolved bug number: –

---

Before merging this pull request to apple/swift repository:
- [ ] Test pull request on Swift continuous integration.

<details>
  <summary>

Triggering Swift CI</summary>

The swift-ci is triggered by writing a comment on this PR addressed to the GitHub user @swift-ci. Different tests will run depending on the specific comment that you use. The currently available comments are:

**Smoke Testing**

| Platform | Comment |
| --- | --- |
| All supported platforms | @swift-ci Please smoke test |
| OS X platform | @swift-ci Please smoke test OS X platform |
| Linux platform | @swift-ci Please smoke test Linux platform |

 **Validation Testing**

| Platform | Comment |
| --- | --- |
| All supported platforms | @swift-ci Please test |
| OS X platform | @swift-ci Please test OS X platform |
| Linux platform | @swift-ci Please test Linux platform |

Note: Only members of the Apple organization can trigger swift-ci.
</details>

### CODE REVIEW

**Comment by eeckstein** [comment]
Do you mind combining the typo-fixes into a single commit (in future)?

**Comment by practicalswift** [comment]
@eeckstein Squashed into three commits :-)

**Comment by gribozavr** [comment]
@swift-ci Please test and merge

**Comment by lattner** [comment]
This patch looks fine to me.

File: include/swift/Runtime/Mutex.h [hunk 1]
```diff
@@ -2,7 +2,7 @@
 //
 // This source file is part of the Swift.org open source project
 //
-// Copyright (c) 2016 Apple Inc. and the Swift project authors
+// Copyright (c) 2014 - 2016 Apple Inc. and the Swift project authors
```
Comment (shawnce) [inline include/swift/Runtime/Mutex.h]: Is it desired / correct to "back date" the copyright for newly added files?

**Comment by lattner** [comment]
I don't think it hurts, and it makes it easier to make sure a file isn't missing a year.

**Comment by shawnce** [comment]
OK, sounds good, will change by behavior for new files

**Pull request merged**